highway/crud_registry.go

```diff
@@ -136,13 +136,27 @@ func (s *HighwayServer) StartAccessName(w http.ResponseWriter, r *http.Request)
 	username := vars["username"]
 
 	// get user
-	whois, err := s.cosmos.QueryName(username)
+	whoisAll, err := s.cosmos.QueryAllNames()
+	var who *rt_v1.WhoIs
+
 	if err != nil {
 		log.Println(err)
 		JsonResponse(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
+	for _, whoIs := range whoisAll {
+		if whoIs.Name == username {
+			who = &whoIs
+		}
+	}
+
+	if who == nil {
+		log.Println(err)
+		JsonResponse(w, errors.New("Error while querying did for domain name"), http.StatusBadRequest)
+		return
+	}
```

Comment:
I updated cosmos.go to perform this logic for querying name, take a look and see if you can just use `s.cosmos.QueryName(username)`

Reply:
Just looked into queryName, and yes i agree. migated from an exhaustive query to the wrapper. Thanks